src/components/Datatable/ControlsModule/ColumnsControls/ColumnsControls.tsx

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+
+import { ControlDropdown } from '../ControlDropdown';
+import ColumnsList from './components/ColumnsList';
+import { useColumnOrder } from './hooks/useColumnOrder';
+
+export interface ColumnsControlsProps {
+  defaultIsColumnsControlsOpen: boolean;
+  onOpen: () => void;
+  onClose: () => void;
+  onApply: (shouldApply: boolean) => void;
+  onReset: () => void;
+  children: (onClick: () => void) => React.ReactElement;
+}
+
+const ColumnsControls: React.FC<ColumnsControlsProps> = ({
+  children,
+  defaultIsColumnsControlsOpen,
+  onOpen,
+  onClose,
+  onApply,
+  onReset,
+}) => {
+  const {
+    orderedColumns,
+    setLocalColumnOrder,
+    storeOrderedColumns,
+    reinitializeOrderedColumns,
+    resetOrderedColumns,
+    isInDefaultOrder,
+  } = useColumnOrder();
+
+  const handleOpenColumnsControl = () => {
+    onOpen();
+  };
+  const handleCloseColumnsControl = () => {
+    onClose();
+    reinitializeOrderedColumns();
+  };
+  const handleApplyColumnsControl = () => {
+    onApply(!isInDefaultOrder);
```

Comment:
Why not do `if (!isInDefaultOrder) onApply()` instead?

Reply:
I need that info to use it in ControlsModule, so this is correct but the intention will be revealed in next PR :)